### PR TITLE
CI environment overhaul

### PR DESCRIPTION
**Pull request by TomAugspurger**
Updates how we install our environment. The goal is to end up with as few subsequent `pip install`  operations as possible. That's what broke our CI over the weekend.

We specify the environments in a per-job `environment.yaml` file. I tried to capture the following cases

1. "Oldest supported". The oldest python, Numpy, pandas, distributed, etc. I'd like to add fastparquet & arrow. Do we have an official minimum version for those @martindurant?
2. "Upstream dev". Upstream versions of pandas, numpy, etc. I'm going to go through the current failures today. I'm hoping to find a better solution to being notified about failures here, since the current system (us checking manually) isn't ideal.

---

Test changes

We should have a few new skips. Some of the fastparquet tests were being skipped, since (IIUC) we previously only tested against fastparquet master.

### CODE REVIEW

File: continuous_integration/travis/install.sh [hunk 1]
```diff
@@ -13,112 +13,39 @@ case "$(uname -s)" in
...
-    pytest-xdist \
-    xxhash \
-    pandas_datareader
+# We don't have a conda-forge package for cityhash
+# We don't include it in the conda environment.yaml, since that may
+# make things harder for contributors that don't have a C++ compiler
+pip install --no-deps cityhash
 
 if [[ ${UPSTREAM_DEV} ]]; then
-    echo "Installing PyArrow dev"
```
Comment (TomAugspurger) [inline continuous_integration/travis/install.sh]: This channel wasn't being updated anymore.

**Comment by TomAugspurger** [comment]
Going to rerun CI a few times to see if any random errors pop up here.

**Comment by TomAugspurger** [comment]
Windows failure on `dask/dataframe/tests/test_shuffle.py::test_temporary_directory`. I can't reproduce locally.

**Comment by TomAugspurger** [comment]
Travis failures were HTTP errors from conda.

File: continuous_integration/appveyor/environment.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: testenv
```
Comment (jrbourbeau) [inline continuous_integration/appveyor/environment.yaml]: I think `toolz` should be explicitly added to the appveyor environment

**Comment by TomAugspurger** [comment]
Thanks @jrbourbeau. Merging.

**Pull request merged**